provider/datagen/src/transform/cldr/locale_canonicalizer/directionality.rs

```diff
@@ -0,0 +1,91 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+use crate::transform::cldr::cldr_serde;
+use icu_locid_transform::provider::*;
+
+use icu_locid_transform::Direction;
+use icu_provider::datagen::IterableDataProvider;
+use icu_provider::prelude::*;
+use std::collections::BTreeMap;
+
+impl DataProvider<DirectionalityV1Marker> for crate::DatagenProvider {
+    fn load(&self, req: DataRequest) -> Result<DataResponse<DirectionalityV1Marker>, DataError> {
+        // We treat searching for `und` as a request for all data. Other requests
+        // are not currently supported.
+        if !req.locale.is_empty() {
+            return Err(DataErrorKind::ExtraneousLocale.into_error());
+        }
+
+        let data: &cldr_serde::directionality::Resource = self
+            .source
+            .cldr()?
+            .core()
+            .read_and_parse("scriptMetadata.json")?;
+        Ok(DataResponse {
+            metadata: Default::default(),
+            payload: Some(DataPayload::from_owned(DirectionalityV1::from(data))),
+        })
+    }
+}
+
+impl IterableDataProvider<DirectionalityV1Marker> for crate::DatagenProvider {
+    fn supported_locales(&self) -> Result<Vec<DataLocale>, DataError> {
+        Ok(vec![Default::default()])
+    }
+}
+
+impl From<&cldr_serde::directionality::Resource> for DirectionalityV1<'_> {
+    fn from(other: &cldr_serde::directionality::Resource) -> Self {
+        let mut map = BTreeMap::new();
+        for (script, metadata) in &other.script_metadata {
+            let rtl = match metadata.rtl {
+                cldr_serde::directionality::Rtl::Yes => Direction::RightToLeft,
+                cldr_serde::directionality::Rtl::No => Direction::LeftToRight,
+                cldr_serde::directionality::Rtl::Unknown => Direction::Unknown,
```

Comment:
As I explained above, `Unknown` and script-not-here is the same thing, so we don't have to store these.